### ISSUE TITLE
[codegen][JAVA] BUG: Ref of a single type ignores min/max and pattern during model generation

### ISSUE THREAD

**Issue by FatCash** [swagger-api/swagger-codegen#8001]
##### Description

If you add a single typed string "object" to pet definition in petstore like below.

The generator does not create any @size annotation and no @pattern.

But when you paste content of nickName instead of $ref, it does create the annotations.
This has to be a bug or am I missing something?

I need to reuse this nickName type in various places and would like to have the min/max constraints defined only once. How do I achieve this?

Note if you create type: object with only one string property then it generates correctly. But it is not appealing to have an objet just for hosting one string.
##### Swagger-codegen version

2.3.1
##### Addition to petstore.yml

```yaml
      nick:
        $ref: "#/definitions/nickName"
  nickName:
    type: string
    minLength: 1
    maxLength: 3
    pattern: "^[A-Z]+$"
```
##### Command line used for generation

swagger-codegen generate -i ./petstore.yml -l spring -o ./petstore

**Comment by FatCash** [comment]
Please, I need help with this so I am rephrasing the question:
This works regarding @size and @pattern generation:
```yaml
      nick:
        type: string
        minLength: 1
        maxLength: 3
        pattern: "^[A-Z]+$"
```
But in this way (as I want), it does not work. Why, how to fix?:
```yaml
      nick:
        $ref: "#/definitions/nickName"
  nickName:
    type: string
    minLength: 1
    maxLength: 3
    pattern: "^[A-Z]+$"
```

**Comment by FatCash** [comment]
Hi, does anyone have a clue how to fix this? Perhaps by modifying some .mustache file?
Can you point me in the right direction, please.

**Comment by FatCash** [comment]
Does anyone have any tips? I get the same behaviour in 3.0.0-SNAPSHOT
@jmini @HugoMario any hints?

**Comment by HugoMario** [comment]
Hey @FatCash i hope to have this address for codegen 3 next week.

**Comment by HugoMario** [comment]
hey @FatCash , can you please check this again, this was fixed for 2.4.0-SNAPSHOT with PR #8277 , still pending fix for 3.0.0 version.

**Comment by iyogi** [comment]
I can confirm that 2.4.0-SNAPSHOT fixes the issue. Can we know when will this version (io.swagger:swagger-codegen-maven-plugin:2.4.0) be released?
